crates/bevy_solari/src/pathtracer/pathtracer.wgsl

```diff
@@ -77,3 +77,36 @@ fn pathtrace(@builtin(global_invocation_id) global_id: vec3<u32>) {
     textureStore(accumulation_texture, global_id.xy, vec4(new_color, old_color.a + 1.0));
     textureStore(view_output, global_id.xy, vec4(new_color, 1.0));
 }
+
+struct NextBounce {
+    wi: vec3<f32>,
+    pdf: f32,
+}
+
+fn importance_sample_next_bounce(wo: vec3<f32>, ray_hit: ResolvedRayHitFull, rng: ptr<function, u32>) -> NextBounce {
+    let diffuse_weight = 1.0 - ray_hit.material.metallic;
+    let specular_weight = ray_hit.material.metallic;
+
+    let TBN = calculate_tbn_mikktspace(ray_hit.world_normal, ray_hit.world_tangent);
+    let T = TBN[0];
+    let B = TBN[1];
+    let N = TBN[2];
+
+    let wo_tangent = vec3(dot(wo, T), dot(wo, B), dot(wo, N));
```

Comment:
If `TBN` is a `mat3<f32>`, this is just `TBN * wo`

Reply:
Hmm this is the code we use everywhere else in bevy 😅